### PR TITLE
fix: [cherry-pick] Only load or release Flushed segment in datanode meta

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #34376 ,  #34375,  #34379

master pr: #34390

### CODE REVIEW

File: internal/datanode/services.go [hunk 1]
```diff
@@ -312,26 +312,35 @@ func (node *DataNode) SyncSegments(ctx context.Context, req *datapb.SyncSegments
...
-		future := io.GetOrCreateStatsPool().Submit(func() (any, error) {
-			var val *metacache.BloomFilterSet
-			var err error
-			err = binlog.DecompressBinLog(storage.StatsBinlog, req.GetCollectionId(), req.GetPartitionId(), newSeg.GetSegmentId(), []*datapb.FieldBinlog{newSeg.GetPkStatsLog()})
-			if err != nil {
-				log.Warn("failed to DecompressBinLog", zap.Error(err))
-				return val, err
+		switch newSeg.GetLevel() {
+		case datapb.SegmentLevel_L0:
+			log.Warn("segment level is legacy, may be the channel has not been successfully watched yet", zap.Int64("segmentID", segID))
```
Comment (congqixia) [inline internal/datanode/services.go:318]: NIT: message is not accurate here

Comment (xiaocai2333) [reply internal/datanode/services.go]: get, fixed it

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**